tests/unit/cocoon/test_cocoon_complex_instrument.py

```diff
@@ -0,0 +1,176 @@
+import unittest
+from pathlib import Path
+import pandas as pd
+
+from lusidtools import cocoon as cocoon
+from parameterized import parameterized
+import lusid
+import lusid.models as models
+from lusid.utilities import ApiClientFactory
+from lusidtools import logger
+from datetime import datetime
+import pytz
+
+
+class ComplexInstrumentTests(unittest.TestCase):
```

Comment:
Great test, can you please add another test to test the failure case where the discriminator isn't valid?